public/src/components/Home/SidebarComponent.jsx

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import ArchivedSubredditsComponent from './ArchivedSubredditsComponent';
+
+const ScrollArea = require('react-scrollbar').default;
+
+export default class SidebarComponent extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            searchFormValue: '',
+            archiveFormValue: '',
+        };
+        this.getSubredditsList = this.getSubredditsList.bind(this);
+        this.onChangeSearchForm = this.onChangeSearchForm.bind(this);
+        this.onChangeArchiveForm = this.onChangeArchiveForm.bind(this);
+        this.getArchiveFormClass = this.getArchiveFormClass.bind(this);
+        this.onKeyPressSearchForm = this.onKeyPressSearchForm.bind(this);
+        this.setSearchValue = this.setSearchValue.bind(this);
+    }
+
+    onKeyPressSearchForm(event) {
+        const { searchFormValue } = this.state;
+        const { storeSubredditToArchive } = this.props;
+        if (searchFormValue && event.key === 'Enter') {
+            storeSubredditToArchive(searchFormValue);
+            this.setState({
+                searchFormValue: '',
+            });
+        }
+    }
+
+    onChangeSearchForm(event) {
+        this.setState({
+            searchFormValue: event.target.value,
+        });
+    }
+
+    onChangeArchiveForm(event) {
+        this.setState({
+            archiveFormValue: event.target.value,
+        });
+    }
+
+    setSearchValue(value) {
+        this.setState({
+            searchFormValue: value,
+        });
+    }
+
+    getArchiveFormClass() {
+        const { subreddits } = this.props;
+        const minimumArchivedLength = 5;
+        const archivedSubreddits = [...subreddits]
+            .filter(subreddit => subreddit.isArchived);
+        return archivedSubreddits.length > minimumArchivedLength
+            ? 'home-sidebar__archive-form'
+            : 'home-sidebar__archive-form disable';
+    }
+
+    getSubredditsList() {
+        const { subreddits } = this.props;
+        const { searchFormValue } = this.state;
+        const subredditList = [...subreddits].map((subreddit, key) => {
+            const formValue = searchFormValue
+                .replace(/\\/g, "\\\\")
+                .toLowerCase()
+                .trim();
+            const subredditTitle = subreddit.display_name
+                .toLowerCase()
+                .trim()
+                .slice(0, subreddit.display_name.length - 1);
```

Comment:
Why do you need to remove the last character?

Reply:
because we want to remove subreddit list after user enters full name of subreddit. 
for example, without this snippet search options will be visible when you enter "askReddit".
But, now, when you enters askReddi, there is no need to show search options

Reply:
fixed https://github.com/YuriyBC/Reddit-archive/commit/bfbb084594f81ad13ebb526a579e5daa1fbcd0e9